### PR TITLE
Add method to adjust note times based on the sustain pedal

### PR DESCRIPTION
**Pull request by e7mac**
Recreating the [apply_sustain_control_changes](https://github.com/magenta/note-seq/blob/484e854b76fb90328a8a42f78014ac3df9a92f50/note_seq/sequences_lib.py#L1517) function from the magenta repo. This is necessary for applying the sustain control to a given note sequence - extending the note end times till the sustain pedal is depressed.

### CODE REVIEW

File: music/src/core/sequences.ts [hunk 1]
```diff
@@ -553,6 +553,179 @@ export function mergeConsecutiveNotes(sequence: INoteSequence) {
...
+        }
+        activeNotes[event.instrument] = newActiveNotes;
+      }
+      // Add this new note to the list of active notes.
+      if (!(event.instrument in activeNotes)) {
+        activeNotes[event.instrument] = [];
+      }
+      activeNotes[event.instrument].push(event);
+    } else if(type === MessageType.NOTE_OFF) {
+      if (susActive[event.instrument] === true) {
```
Comment (notwaldorf) [inline music/src/core/sequences.ts:706]: indenting is a bit whack here

**Review by notwaldorf — COMMENTED** [review]
@iansimon would you mind taking a look at this when you have a chance? I think you're the sequences expert 🥰

**Comment by iansimon** [comment]
> @iansimon would you mind taking a look at this when you have a chance? I think you're the sequences expert 🥰

Weird, I took a look at it a few days ago but the Review button wasn't showing up for me.  Anyway it looks good to me.

**Review by iansimon — APPROVED** [review]

File: music/src/core/sequences.ts (hunk 1 above)
Comment (e7mac) [reply music/src/core/sequences.ts:706]: pushed a commit fixing the spacing

**Review by notwaldorf — APPROVED** [review]

**Comment by notwaldorf** [comment]
(Published as 1.23.0)

**Pull request merged**